File: src/main/java/com/googongill/aditory/domain/Link.java
package com.googongill.aditory.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Link extends BaseEntity {
    @Id @Column(name = "link_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String summary;
    private String url;
    private Boolean linkState;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    public Link(String title, String summary, String url, Category category, User user) {
        this.title = title;
        this.summary = summary;
        this.url = url;
        this.linkState = false;
        this.category = category;
        this.user = user;
    }

    public Link(String title, String url, Boolean linkState, Category category, User user) {
        this.title = title;
        this.url = url;
        this.linkState = linkState;
        this.category = category;
        this.user = user;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public void updateLinkInfo(String title, String summary, String url, Category category) {
        this.title = title;
        this.summary = summary;
        this.url = url;
        this.category = category;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public void updateLinkState() {
        this.linkState = true;
    }
}

File: src/main/java/com/googongill/aditory/domain/Link.kt
package com.googongill.aditory.domain

import jakarta.persistence.*

@Entity
class Link : BaseEntity {
    @Id
    @Column(name = "link_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    var title: String
    var summary: String? = null
    var url: String
    var linkState: Boolean

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    var category: Category

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: User

    constructor(title: String, summary: String?, url: String, category: Category, user: User) {
        this.title = title
        this.summary = summary
        this.url = url
        linkState = false
        this.category = category
        this.user = user
    }

    constructor(title: String, url: String, linkState: Boolean, category: Category, user: User) {
        this.title = title
        this.url = url
        this.linkState = linkState
        this.category = category
        this.user = user
    }

    // 연관관계 메서드
    fun updateCategory(category: Category) {
        this.category = category
    }

    fun updateLinkInfo(title: String, summary: String?, url: String, category: Category) {
        this.title = title
        this.summary = summary
        this.url = url
        this.category = category
    }

    fun updateUser(user: User) {
        this.user = user
    }

    fun updateLinkState() {
        linkState = true
    }
}
